### PR TITLE
docs: release v0.14.0

### PR DESCRIPTION
**Pull request by jm-clius**
Adds the release notes for [Release v0.14.0](https://github.com/status-im/nwaku/milestone/6)

I believe everything necessary for the release has been merged already. I will do a last sanity check before merging this PR to close the last open issues.

For the items you were responsible for, please check:

- that it's included in the release notes
- that you're happy with the phrasing

### CODE REVIEW

**Review by LNSD — APPROVED** [review]
LGTM ✅🚀

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,68 @@
...
+
+Release highlights:
+- An important fix for the Waku message archive returning inconsistent responses to history queries.
+- Support for [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/) and [libp2p Circuit Relay](https://docs.libp2p.io/concepts/nat/circuit-relay/) that allows, among other things, for [NAT hole punching](https://docs.libp2p.io/concepts/nat/hole-punching/).
+- Support for structured logging in JSON format.
+- A fix for an underlying file descriptor leak that affected websocket connections.
+
+### Features
+
+- Support for [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/)
```
Comment (alrevuelta) [inline CHANGELOG.md:11]: just a preference, but like having a link to the PR to see the changes (like in the fixes section)

- Support for [AutoNAT](https://docs.libp2p.io/concepts/nat/autonat/) [#1472](https://github.com/waku-org/nwaku/pull/1472)

Not sure if you could try the [autogenerated](https://github.com/waku-org/nwaku/pull/1378#discussion_r1021893395) notes. With good PR namings it makes creating the release notes quite easy :D

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,68 @@
...
+- New Waku Archive implementation. This allows easy addition of drivers for different technologies to store historical messages.
+- Support for structured logging and specifying log format.
+- Node now keeps track of its external reachability.
+
+### Changes
+
+- Zerokit RLN library now statically linked.
+- Use extended key generation in Zerokit API to comply with [32/RLN](https://rfc.vac.dev/spec/32/).
+- Re-enable root validation in [`17/WAKU-RLN-RELAY`](https://rfc.vac.dev/spec/17/) implementation.
+- [Network monitoring tool](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/tools#networkmonitor) now supports DNS discovery.
```
Comment (alrevuelta) [inline CHANGELOG.md:22]: just a preference.

[Network monitoring tool](https://github.com/status-im/nwaku/tree/2336522d7f478337237a5a4ec8c5702fb4babc7d/tools#networkmonitor) now supports DNS discovery. [#1446](https://github.com/waku-org/nwaku/pull/1446)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,68 @@
...
+- Peerstore now consolidated with libp2p peerstore.
+- Peerstore now also tracks peer direction.
+- SIGSEGV signals are now handled and logged properly.
+- Waku v2 no longer imports libraries from Waku v1.
+- Improved build and CI processes:
+  - Added support for an `EXPERIMENTAL` compiler flag.
+  - Simplified project Makefile.
+  - Split Dockerfile into production and experimental stages.
+  - Removed obsolete simulation libraries from build.
+- Improved parallellisation (and therefore processing time) when dialing several peers simultaneously.
```
Comment (alrevuelta) [inline CHANGELOG.md:35]: Improved parallellisation (and therefore processing time) when dialing several peers simultaneously. [#1471](https://github.com/waku-org/nwaku/pull/1471)

**Review by alrevuelta — APPROVED** [review]
lgtm! just a non-blocking opinion in case you want to revisit [this](https://github.com/waku-org/nwaku/pull/1378#discussion_r1021893395).

**Review by rymnc — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (jm-clius) [reply CHANGELOG.md:11]: Thanks! I do use the autogenerated notes as starting point, but find that to make it humanly interpretable requires significant rewrite (e.g. many "features" and "fixes" just fall under a "updated submodules"). It was a bit easier to rewrite and keep links to PRs related to fixes. I think this is the way to go in the future when we have automated releases, but we may need to revisit how commit messages are formatted.

Comment (alrevuelta) [reply CHANGELOG.md:11]: sure, not suggesting to use the autogenerated notes by default. +1 for human readable.
my only concern was to link the PR of each feature :)

**Pull request merged**